docs/root/configuration/http/http_filters/header_mutation_filter.rst

```diff
@@ -0,0 +1,25 @@
+.. _config_http_filters_header_mutation:
+
+Header Mutation
+===============
+
+* This filter should be configured with the type URL ``type.googleapis.com/envoy.extensions.filters.http.header_mutation.v3.HeaderMutation``.
+* :ref:`v3 API reference <envoy_v3_api_msg_extensions.filters.http.header_mutation.v3.HeaderMutation>`
+
+This is a filter that can be used to add, remove, append, or update HTTP headers. It can be added in any position in the filter chain
+and used as downstream or upstream HTTP filter. The filter can be configured to apply the header mutations to the request, response, or both.
+
+
+In most cases, this filter would be a more flexible alternative to the ``request_headers_to_add``, ``request_headers_to_remove``,
```

Comment:
Is it fair to say this alternative use would sometimes also be done via composite filter?

Reply:
What do you mean by the `composite filter` here? The `envoy.filters.http.composite` filter or multiple header mutation filters?

Reply:
envoy.filters.http.composite

Reply:
I think the answer is yes. The compositer filter could provides very flexible matching. Although for most users, route level configuration is flexible enough.